sklearn/cluster/_spectral.py

```diff
@@ -258,9 +258,14 @@ def spectral_clustering(
       Ulrike von Luxburg
       http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.165.9323
 
-    - Multiclass spectral clustering, 2003
```

Comment:
This paper seems relevant since this is the method implemented to assign the labels with `assign_labels="discretize"`. It is indeed the implementation of the `discretize` function on the top of the file. We could be more explicit in the documentation to refer to this paper in the section of the `assign_labels`.

Reply:
We should improve the `References` section with numbering to have a clear idea of which paper is relevant for which parameter and part of the implemented algorithm.

Reply:
@glemaitre this paper is actually referenced 3 times in this file. Other papers are referenced twice, just repeated. Is it the way referencing is intended, with duplications?

Reply:
It is fine to have duplication. Most of the time, our user will look at the documentation generated from the docstring and not the source. It means that if one uses the `discretize` function, it should be documented. Similarly, if one uses the `SpectralClustering`, it is meaningful to have the reference to know what corresponds to the `"discretize"` option.

Reply:
OK, will do